brcm_fwpatch_ldr.py

```diff
@@ -14,15 +14,18 @@ def accept_file(li, n):
         pass
     return 0
 
-def make_seg(addr_range):
-    seg = idaapi.segment_t()
-    seg.bitness = 1 # 32bit
-    seg.startEA = addr_range[0]
-    seg.endEA = addr_range[1]
-    seg.perm = idaapi.SEGPERM_READ | idaapi.SEGPERM_WRITE | idaapi.SEGPERM_EXEC
-    seg.type = idaapi.SEG_CODE
-    # use bss to tell what is not patched
-    idaapi.add_segm_ex(seg, 'patchram', 'BSS', idaapi.ADDSEG_NOSREG)
+def make_seg(addr_range, record_type = 0):
+        seg = idaapi.segment_t()
+        seg.bitness = 1 # 32bit
+        seg.startEA = addr_range[0]
+        seg.endEA = addr_range[1]
+        seg.perm = idaapi.SEGPERM_READ | idaapi.SEGPERM_WRITE | idaapi.SEGPERM_EXEC
+        seg.type = idaapi.SEG_CODE
+        # use bss to tell what is not patched
+        idaapi.add_segm_ex(seg, 'patchram', 'BSS', idaapi.ADDSEG_NOSREG)
+        if record_type == 0x08:
```

Comment:
looks like it can just be:
```python
seg.perm = idaapi.SEGPERM_READ | idaapi.SEGPERM_EXEC
if record_type != 0x08:
  seg.perm |= idaapi.SEGPERM_WRITE
idaapi.add_segm_ex(seg, 'patchram', 'BSS', idaapi.ADDSEG_NOSREG)
```
btw, what is type 8 used for?

Reply:
Hmm.

What if it does the following:
rom1:
Create seg with addr: x0 size: xC8000
Asks to load additional binary and then use patch_many_bytes with 0x08 records,
else empty bytes

rom2:
Create seg with addr: 0x260000 size: xC000
Asks to load additional binary,
else empty bytes

ram_lo:
Create seg with addr: 0x0D0000 size: x10000
ram_hi:
Create seg with addr: 0x200000 size: x48000

And then use patch_many_bytes with 0x0A records to patch ram_lo and ram_hi.

So basically it will create the basic segments and then user can identify the patchram data by patched bytes.

-----------

BTW, the above, will work correctly when we get the unpatched ROM, by using a patchram with 0x08 stripped (except if it can boot with no patchram). 
I would do that, but I don't know if the device will boot in a working state.
And I don't have the spi programmer yet, so I don't want to lose a 2nd Joy-Con. 

Lastly, it seems that sub_6EC0E handles patchram.

Reply:
That sounds good, now that memory layout is known :)